lib/features/support/header.rb

```diff
@@ -0,0 +1,25 @@
+require 'digest/sha1'
+require 'json'
+
+def valid_bugsnag_integrity_header(request)
+  valid = valid_sha_digest_header(request) || valid_simple_digest_header(request)
+  assert_true(valid, "Bugsnag-Integrity header does not match SHA1 or Request Length digest.")
+end
+
+def valid_sha_digest_header(request)
+  header = request[:request]["Bugsnag-Integrity"]
+  assert_not_nil(request[:body], "No request body present")
+  body = JSON.generate(request[:body])
```

Comment:
Computing the digest after Maze Runner has parsed JSON text into a Ruby hash, and this has generated it back into JSON, isn't going to work.  Any differences in whitespace will lead to a difference in the digest and it needs to be computed "at source".

Perhaps if you can action the other comments, then assign the ticket to me and I'll put that in though.

Reply:
> Computing the digest after Maze Runner has parsed JSON text into a Ruby hash, and this has generated it back into JSON, isn't going to work. Any differences in whitespace will lead to a difference in the digest and it needs to be computed "at source".

I think we still need to do this before merging

Reply:
Well spotted, that almost slipped through the net.  I've made some changes to the steps and unit tests, but will rerun the Android test that relies on it as well to ensure that it's what is required.